fetcher/fetcher_model.go

```diff
@@ -0,0 +1,204 @@
+package fetcher
+
+import (
+	"encoding/xml"
+	"fmt"
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+// RepoMd has repomd data
+type RepoMd struct {
+	RepoList []Repo `xml:"data"`
+}
+
+// Repo has a repo data
+type Repo struct {
+	Type     string   `xml:"type,attr"`
+	Location Location `xml:"location"`
+}
+
+// Location has a location of repomd
+type Location struct {
+	Href string `xml:"href,attr"`
+}
+
+// Reference has reference information
+type Reference struct {
+	Href  string `xml:"href,attr" json:"href,omitempty"`
+	ID    string `xml:"id,attr" json:"id,omitempty"`
+	Title string `xml:"title,attr" json:"title,omitempty"`
+	Type  string `xml:"type,attr" json:"type,omitempty"`
+}
+
+// Package has affected package information
+type Package struct {
+	Name     string `xml:"name,attr" json:"name,omitempty"`
+	Epoch    string `xml:"epoch,attr" json:"epoch,omitempty"`
+	Version  string `xml:"version,attr" json:"version,omitempty"`
+	Release  string `xml:"release,attr" json:"release,omitempty"`
+	Arch     string `xml:"arch,attr" json:"arch,omitempty"`
+	Filename string `xml:"filename" json:"filename,omitempty"`
+}
+
+// uniquePackages returns deduplicated []Package by Filename
+// If Filename is the same, all other information is considered to be the same
+func uniquePackages(p []Package) []Package {
+	tmp := make(map[string]Package)
+	ret := []Package{}
+	for _, pac := range p {
+		tmp[pac.Filename] = pac
+	}
```

Comment:
The argument type is []Package, so how about ps or pkgs?
```suggestion
func uniquePackages(pkgs []Package) []Package {
	tmp := make(map[string]Package)
	ret := []Package{}
	for _, pkg := range pkgs {
		tmp[pkg.Filename] = pkg
	}
```